pulsar-client-cpp/README.md

```diff
@@ -33,34 +40,84 @@ sudo cp *.a /usr/lib
 ```
 
 
-Compile Pulsar client library:
+#### Compile Pulsar client library:
 
 ```shell
 cd pulsar/pulsar-client-cpp
 cmake .
 make
 ```
 
-Client library will be placed in
+#### Checks
+##### Client library will be placed in
 ```
 lib/libpulsar.so
 lib/libpulsar.a
 ```
 
-Tools :
+##### Tools will be placed in
 
 ```
 perf/perfProducer
 perf/perfConsumer
 ```
 
-Tests:
+### Compile on Mac OS X
+
+#### Install all dependencies:
+```shell
+# For openSSL
+brew install openssl
+
+# For Protobuf
+brew tap homebrew/versions
+brew install protobuf260
+
+# For gtest
+cd $HOME
+git clone https://github.com/google/googletest.git
+cd googletest
+cmake .
+make 
+## Refer gtest documentation in case you get stuck somewhere
+export PATH=$HOME/googletest/build/googlemock/gtest/:$HOME/googletest/googletest/include/:$PATH
+```
+
+#### Compile Pulsar client library:
+```shell
+export $PULSAR_PATH=<Path where you cloned pulsar repo>
+cd ${PULSAR_PATH}/pulsar-client-cpp/
+cmake .
+make
+```
+
+#### Checks
+##### Client library will be placed in
+```
+${PULSAR_PATH}/pulsar-client-cpp/lib/libpulsar.dylib
+${PULSAR_PATH}/pulsar-client-cpp/lib/libpulsar.a
+```
+
+##### Tools will be placed in:
 
 ```
- 1. Start the standalone pulsar
-    export PULSAR_STANDALONE_CONF=$(pwd)/tests/standalone.conf
-    ../bin/pulsar standalone
+${PULSAR_PATH}/pulsar-client-cpp/perf/perfProducer
+${PULSAR_PATH}/pulsar-client-cpp/perf/perfConsumer
+```
 
- 2. Run tests
-    tests/main
+## Tests
 ```
+# Source code
+${PULSAR_PATH}/pulsar-client-cpp/tests/
+
+# Execution
+## Start standalone broker
+export PULSAR_STANDALONE_CONF=${PULSAR_PATH}/pulsar-client-cpp/tests/standalone.conf
+${PULSAR_PATH}/bin/pulsar standalone
+
+## Run the tests
+${PULSAR_PATH}/pulsar-client-cpp/tests/main
+```
+
+## Requirements for Contributors
```

Comment:
Will change the travis build script and add the setup script for mac as a part of a separate pull request.